test/rekt/features/containersource/features.go

```diff
@@ -118,3 +118,27 @@ func SendsEventsWithArgs() *feature.Feature {
 
 	return f
 }
+
+func SendEventsWithTLSRecieverAsSink() *feature.Feature {
+	source := feature.MakeRandomK8sName("containersource")
+	sink := feature.MakeRandomK8sName("sink")
+	f := feature.NewFeature()
+
+	f.Setup("install sink", eventshub.Install(sink, eventshub.StartReceiverTLS))
+
+	f.Requirement("install ContainerSource", func(ctx context.Context, t feature.T) {
+		d := service.AsDestinationRef(sink)
+		d.CACerts = eventshub.GetCaCerts(ctx)
+
+		containersource.Install(source, containersource.WithSink(d))(ctx, t)
+	})
+	f.Requirement("containersource goes ready", containersource.IsReady(source))
+
+	f.Stable("containersource as event source").
+		Must("delivers events",
+			assert.OnStore(sink).MatchEvent(
+				test.HasType("dev.knative.eventing.samples.heartbeat"),
```

Comment:
TLS tests and your PR made me realize this: https://github.com/knative-sandbox/reconciler-test/pull/535, so technically, this match might match events that are "rejected" so travelling in plaintext on non TLS connection, therefore, for the time being, we need the MatchKind call, similar to APIServerSource test:

https://github.com/knative/eventing/blob/5e245ac60ff1fc5a77976501713f60ba1e8f9940/test/rekt/features/apiserversource/data_plane.go#L203-L206

Reply:
I added the matching, all the events were of type "rejected"

Reply:
Ok, that's a nice finding, I believe we need to add CACerts to the SinkBinding resources (similar to changes in config/core/resources/containersource.yaml) since ContainerSource uses SinkBinding under the hood, so if CACerts is not propagated to it, it won't work